CMakeLists.txt

```diff
@@ -1,93 +1,41 @@
-######### CMake Version #####################
-cmake_minimum_required(VERSION 2.8.11)
-#############################################
+project(eigen-cuda LANGUAGES CXX)
+cmake_minimum_required(VERSION 3.14)
 
-######### Options ###########################
-option( CORE_USE_CUDA    "Use CUDA to speed up certain parts of the code."              ON )
-#############################################
+option(USE_CUDA "Whether to use CUDA" OFF)
 
-
-######### CUDA decisions ####################
-if (CORE_USE_CUDA)
-    MESSAGE( STATUS ">> --------------  USING CUDA --------------" )
-    set( CUDA_TOOLKIT_ROOT_DIR "/opt/cuda" )
-    if (APPLE OR UNIX)
-        set(CMAKE_C_COMPILER   /opt/cuda/bin/gcc)
-        set(CMAKE_CXX_COMPILER /opt/cuda/bin/g++)
-    elseif (WIN32)
-        ### By default we use VS
-        MESSAGE( STATUS ">> User compiler:             MSVC" )
-        MESSAGE( STATUS ">> Choosing a different compiler is not yet implemented for Windows" )
-    endif()
-endif()
-#############################################
-
-
-######### Info ##############################
-MESSAGE( STATUS ">> CMAKE_C_COMPILER:        " ${CMAKE_C_COMPILER} )
-MESSAGE( STATUS ">> CMAKE_CXX_COMPILER:      " ${CMAKE_CXX_COMPILER} )
-#############################################
-
-
-######### Project Name ######################
-project(eigencuda)
-SET( EXECUTABLE_NAME run )
-#############################################
-
-
-### Find includes in corresponding build directories
-set( CMAKE_INCLUDE_CURRENT_DIR ON )
-######### Have the binary placed into the source head
-set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR} )
-### Output paths for multi-config builds (e.g. msvc)
-foreach( OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES} )
-    string( TOUPPER ${OUTPUTCONFIG} OUTPUTCONFIG )
-    set( CMAKE_RUNTIME_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${CMAKE_SOURCE_DIR} )
-    # set( CMAKE_LIBRARY_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${youroutputdirectory} )
-    # set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${youroutputdirectory} )
-endforeach( OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES )
-#############################################
-set( CMAKE_DISABLE_SOURCE_CHANGES  ON )
-set( CMAKE_DISABLE_IN_SOURCE_BUILD ON )
```

Comment:
Why did you remove these two lines, have the defaults in cmake changed?

Reply:
sry I wasn't aware of these cmake options, which seem to be undocumented https://stackoverflow.com/questions/1208681/with-cmake-how-would-you-disable-in-source-builds. Doesn't hurt to add this here, added